runtime/runtime.go

```diff
@@ -786,7 +802,25 @@ func validateArgumentParams(
 			}
 		}
 
-		arg, err := importValue(inter, value, parameterType)
+		var arg interpreter.Value
+		panicError := panicToError(func() {
+			// if importing an invalid public key, this call panics
+			arg, err = importValue(inter, value, parameterType)
+		})
+
+		if panicError != nil {
+			// The only potential user error here is InvalidPublicKeyError,
+			// so panic if any other error occurs.
+			if _, ok := panicError.(interpreter.InvalidPublicKeyError); ok {
```

Comment:
Why do we have to special case `interpreter.InvalidPublicKeyError` here? Maybe just do it for any recovered error

Reply:
My understanding is that we only want to turn user-caused panics into errors. Here that should only be `InvalidPublicKeyError` so everything else should actually panic.

Would it make more sense to wrap everything that isn't a go panic? Like how `wrapPanic` does it but returning an error instead of panicking with an `ExternalError`.

Reply:
I went with keeping `panicToError` since it seems like a good function for composability. Then I call it within `userPanicToError`, which panics at go errors while returning every other error. So no more special case for `InvalidPublicKeyError`.